.github/workflows/CI-using-ghub-reg.yaml

```diff
@@ -0,0 +1,44 @@
+name: CI using github registry for dev.
+
+on:
+  pull_request:
+    branches: 
+      - dev
+  push:
+    branches: 
+      - dev
+
+env:
+  REGISTRY: ghcr.io
+  IMAGE_NAME: ${{ github.repository }}
+
+jobs:
+  build-and-push-image:
+    runs-on: ubuntu-latest
+    permissions:
+      packages: write
+      contents: read
+
```

Comment:
What is this file addition for? 
I'm sure it is necessary but please state why.

Reply:
@ranwahle Just FYI, this is the file I created for the CI/CD using Github Action, it exists only on the `dev` branch at the moment, so any merge request to the `main` branch -  will show this as an added file.